### DISCUSSION TITLE
GitHub Packages | Support for python packages

### DISCUSSION THREAD

**Discussion by TarunavBA** [community/community#8542] [Packages]
Right now you could only upload to pypi or leave it like that. I want to upload my packages to GitHub packages just like docker, npm, gradle, maven etc. It would be awesome if you could release it in the public beta where users can simply go to feature preview and enable it.

**Comment by DannyVarod** [comment]
This has been a request for a few years now:
https://github.community/t/pypi-compatible-github-package-registry/14615
Still no commitment to this though:
https://github.com/orgs/github/projects/4247/views/1?filterQuery=pypi

**Reply by codebydant** [reply]
Does anyone know how to report this to a GitHub Developer?

**Reply by lukasz-mitka** [reply]
And now it's no longer planned:
https://github.com/github/roadmap/issues/94#issuecomment-1581086839
> This is no longer planned due to a change in our strategic priorities and the allocation of our resources towards higher-priority initiatives.

**Reply by rwaal** [reply]
This is really disappointing. Given how popular Python has become, this decision is pretty surprising. Looks like we'll need to consider alternatives like Artifactory.

**Reply by bodograumann** [reply]
Agreed. Especially seeing how simple the pypi repo api is.
From what I remember, when looking into this, you only need a static http server.
Cf. https://bartbroere.eu/2020/04/07/running-pypi-registry-windows-github-actions/
The main difficulty would be authentication with a private registry, which I didn't find any existing solution for.

Alternatively it is possible to install packages directly through git access:
Cf. https://www.freecodecamp.org/news/how-to-use-github-as-a-pypi-server-1c3b0d07db2/

**Reply by jstafford5380** [reply]
Or... prepare for amusement... Azure Artifacts. It has supported PyPi forever.

**Reply by chicco785** [reply]
> This is no longer planned due to a change in our strategic priorities and the allocation of our resources towards higher-priority initiatives.

Come on... did they really come up with such a sentence? Such a corporate cliché...

**Reply by lukasz-mitka** [reply]
Hey guys, don't shoot the messenger 😆
![image](https://github.com/user-attachments/assets/f3f97a01-ce02-4346-be08-c446ee792672)

**Reply by fdcastel** [reply]
> This is no longer planned due to a change in our strategic priorities and the allocation of our resources towards higher-priority initiatives.

AI generated content?

Python is **very** active at Microsoft. I'm frankly surprised this issue hasn't gained traction yet.

**Reply by dhvcc** [reply]
> > This is no longer planned due to a change in our strategic priorities and the allocation of our resources towards higher-priority initiatives.
> 
> AI generated content?
> 
> Python is **very** active at Microsoft. I'm frankly surprised this issue hasn't gained traction yet.

Yep, pretty much

Own nothing and be happy + use AI for everything and be happy

Who needs packages if AI can regurgitate and spew out the same package internals, but worse and directly in your own codebase 😆

**Comment by ashrafguitoni** [comment]
It's really unfortunate that such a helpful feature is still not scheduled for development yet. Really hoping to see this in GitHub Beta in the near future!

**Reply by davidkhala** [reply]
This has been added to [roadmap](https://github.com/github/roadmap/issues/94)

**Reply by ashrafguitoni** [reply]
@davidkhala I think it's been in `Future` for nearly 2 years. Any chance it'll be moved to one of the other boards?

**Comment by pulkitaggarwl** [comment]
I agree this is a great value add to the product, we do have this on our roadmap and we will try to get to near future. Unfortunately no ETA at the moment.

**Reply by luwangshell** [reply]
Our company is also requiring this feature as priority.

**Reply by ghost** [reply]
my company too

**Comment by hgad** [comment]
Really important to have this one for all the Python shops out there.

**Comment by timmypidashev** [comment]
Would love this feature so much <3

**Comment by xqi-aviatrix** [comment]
Python packages would be very helpful

**Comment by csmizzle** [comment]
would love this feature so much!

**Comment by Clean-Cole** [comment]
GitLab has had this available for at least a year already. Would be nice to see in GitHub as well.

**Comment by vinicius-roc** [comment]
this feature would so much amazing! We need too!

**Reply by hannesdelbeke** [reply]
can't wait :)

**Comment by leovanzella** [comment]
Would be great having this feature!

**Comment by gerardsimons** [comment]
Please, please add this !

**Comment by danielnyari** [comment]
It's on the [public roadmap](https://github.com/github/roadmap/issues/94). There is no date for it though. Please make this happen 🙏

**Reply by ashrafguitoni** [reply]
I don't know if this post (given its age) is being monitored by anyone from the Github team ...

**Reply by timmypidashev** [reply]
It really should be given the fact that packaging Python packages has become so common and standard in many projects. Providing a proper, official way to do this will boost Github's packaging support 👍> I don't know if this post (given its age) is being monitored by anyone from the Github team ...

**Comment by codebydant** [comment]
This a superb idea, this will help my organization to not depend on external services such as JFROG and keep all our artifacts in one place.

**Comment by codebydant** [comment]
Hi folks,

I have contacted GitHub Support asking for this open discussion and this was the formal response.

![Screenshot from 2023-01-03 13-02-51](https://user-images.githubusercontent.com/115469901/210415131-cb288a55-3f2f-4616-922e-a8f6079a3093.png)

**Reply by ashrafguitoni** [reply]
Thank you for taking the time to do this. It's unfortunate that this won't be happening soon. Maybe in 2024 ...

**Reply by alberto-corrales** [reply]
Maybe in 2025.... :(

**Reply by yunjie17** [reply]
Maybe in 2026....:(

**Reply by yelsayed** [reply]
Maybe in 2027... :'(

**Comment by mohas556** [comment]
This is a fundamentally critical piece that is preventing a lot of companies from migrating away from GitLab, which has an excellent implementation of this. Quite shocking that Engineering is so blasé about it .

**Reply by oh54** [reply]
Couldn't agree more. 
Especially given the fast pace of developments in data science and machine learning, where Python is the de facto standard.

Microsoft, this is a must have for many data science teams before moving their codebase to GitHub!

**Reply by dpgraham4401** [reply]
looking at this issue for this exact reason

**Comment by Seluj78** [comment]
Any news on this ? It's critical for many of us

**Comment by nearlythere** [comment]
Is there any news or update about this? In Q3 2020 the python package repo was showing as Q1 2021. In Q4 it moved to future and has been there ever since. [GitHub public roadmap • github](https://github.com/orgs/github/projects/4247/views/1?filterQuery=pypi)

**Reply by timmypidashev** [reply]
This feature would be pretty cool, I've been keeping an eye on this issue for almost a year now and nothing new unfortunately :(

**Reply by lukasz-mitka** [reply]
https://github.com/github/roadmap/issues/94#issuecomment-1581086839
> This is no longer planned due to a change in our strategic priorities and the allocation of our resources towards higher-priority initiatives.

**Comment by aslanpnahe4** [comment]
ُسلام

**Comment by BenjaminYde** [comment]
Any more news on this?
On the github roadmap this is in Future column.
This feature would be very nice to have... 
![image](https://user-images.githubusercontent.com/44135938/233962442-c4790add-53b5-4d90-a91e-b9ed18300b21.png)

**Reply by lukasz-mitka** [reply]
https://github.com/github/roadmap/issues/94#issuecomment-1581086839
> This is no longer planned due to a change in our strategic priorities and the allocation of our resources towards higher-priority initiatives.

**Comment by BeZamorano** [comment]
Any updates on this one?

**Reply by lukasz-mitka** [reply]
https://github.com/github/roadmap/issues/94#issuecomment-1581086839
> This is no longer planned due to a change in our strategic priorities and the allocation of our resources towards higher-priority initiatives.

**Comment by lukasz-mitka** [comment]
https://github.com/github/roadmap/issues/94#issuecomment-1581086839
> This is no longer planned due to a change in our strategic priorities and the allocation of our resources towards higher-priority initiatives.

**Reply by bodograumann** [reply]
Just seen it. Couldn't even react on the roadmap issue.
This is really disappointing...

**Reply by dhvcc** [reply]
Wow, that's really... Interesting
Having Guido work at Microsoft and their focus on all the AI stuff, but still refusing to push python ecosystem

**Comment by ObserverOfTime** [comment]
https://docs.gitlab.com/ee/user/packages/pypi_repository/ :clap: 
https://docs.gitea.com/packages/packages/pypi :clap:

**Comment by subesokun** [comment]
Looks like they've canceled the other pending package features for the same reason. https://github.com/github/roadmap/issues?q=is%3Aissue+allocation

**Comment by BenjaminYde** [comment]
Why is this not supported anymore?
 This is a basic feature that is supported by every platform?
 Will this come back in the future?

**Comment by techdragon** [comment]
Its hard to ignore the possibility that they were told to stop all further packaging work by their corporate masters at Microsoft ... Seeing as further development of Github packages competes with Azure Artefacts and Microsoft are all in on Azure. https://azure.microsoft.com/en-us/products/devops/artifacts/  

Their quoted reasons of a "change in our strategic priorities" would fit nicely into corporate management speak for "the executives don't want us competing with other internal products"

**Reply by 12rambau** [reply]
I am currently discovering devOps at my company and I'm .... disapointed by the tool to say the least. I proposed to my boss to migrate to a Github private organization and the only missing piece is this. So yes, maybe they want to keep devOps alive more than it should. 

But look what I found: https://github.com/astariul/github-hosted-pypi

**Reply by dhvcc** [reply]
@12rambau at this point you can use GitLab and save your company time with dealing with GitHub ops

**Reply by BenjaminYde** [reply]
For package management we use devpi to host our python packages.
We moved from Azure to Github half a year ago. 
It is much more simple to use for all of our developers.
We also use Azure, but not the AzureDevOps part

**Reply by 12rambau** [reply]
thanks for your comments but my company is tied up with Microsoft tools so I don't see them move an inch because the new GIS expert wants to change the dev infra ;-)

**Comment by shayms8** [comment]
How come its not supported yet? :( 
very disappointing, especially because GitHub team said its on the roadmap but now they canceled it.
Or am i missing something? is it planned for the future?

as @ObserverOfTime mentioned, seems like it exists on https://docs.gitlab.com/ee/user/packages/pypi_repository/

**Reply by lukasz-mitka** [reply]
At this time it's no longer planned. See https://github.com/github/roadmap/issues/94#issuecomment-1581086839

**Comment by Sierrapup11** [comment]
-

**Comment by will-e-yams** [comment]
Welcome to the time capsule! It's now just a few weeks before 2024 and the community would like you to know that this is still a very desirable feature!!! Thanks, GitHub!!!

**Reply by lukasz-mitka** [reply]
It's currently top Packages Discussion (by a lot) and that still doesn't mean anything to GitHub.

**Reply by RobertAlbus** [reply]
Disappointing to say the least. My company considered migrating from JFrog Artifactory Cloud, but not having Python support makes it a non-starter. We see TBs of data transfer in packages every month and many tens of thousands of package artifacts would be added to our Enterprise Cloud subscription. Not a small customer.

**Comment by RempelOliveira** [comment]
Does anyone know if it's possible to dockerize a python package and import it into another project?

I ask this because, there is a container registry for pushing dockerized images in the github packages resource.

Is this a possible easy way to do this?

**Reply by lukasz-mitka** [reply]
Not directly related to your question, but I'll add it just in case.

You can install pip packages directly from the git repo https://pip.pypa.io/en/stable/topics/vcs-support/

**Reply by noirbizarre** [reply]
Only works for source-only packages. If you need to build anything, it means you need to have a toolchain where you install which is not very safe in production, not very scalable and error prone.

**Reply by BenjaminYde** [reply]
1. Create an azure vm with a docker compose using devpi, nginx and certbot
2. Build and upload packages to devpi
3. In dockerfile, install your python package from devpi instead of pypi (Make sure to login to devpi using devpi-tokens)

**Comment by sourceful-karlson** [comment]
https://github.com/astariul/github-hosted-pypi
https://github.com/astariul/github-hosted-pypi

**Reply by 12rambau** [reply]
Don't you think it's a bit aggressive to send it 8 times? 😅

**Reply by sourceful-karlson** [reply]
edited!

**Reply by abrichr** [reply]
@sourceful-karlson does this solve the problem of not having anything listed under "Packages" in a Github repo that publishes to PyPi?

<img width="315" alt="image" src="https://github.com/community/community/assets/774615/a5e86f0e-b92b-4d99-ade3-a51cea63b28e">

**Comment by RakeshNagarajan** [comment]
Any update on this request?

**Comment by umm0n** [comment]
Unbelievable this isn't available yet...

**Comment by RyanGuild-us-ignite** [comment]
Hi, I'm just another GitHub user—maybe even a zealot. I need GitHub to support something that allows me to separate sources from compiled artifacts in Earth's second most popular programming language. Thank you

**Comment by mikeball** [comment]
I also need to host internal python packages and would really like to use github actions for ci/cd. Surely would be nice to just use the same platform that I'm already using for docker containers and npm packages. How on earth can this not a like a top priority for the most popular language?

**Comment by BenjaminYde** [comment]
They are collaborating with JFrog (Artifactory) so I don't think this will be ever a thing.
See the blogpost: https://github.blog/2024-05-29-github-and-jfrog/
This is a feature that gitlab, azure devops, ... has, why not Github? Python is one of the most common used languages.

**Comment by ei-grad** [comment]
https://github.com/github/roadmap/issues/94
> This is no longer planned due to a change in our strategic priorities and the allocation of our resources towards higher-priority initiatives.

Should our higher-priority initiative be to move our repos, CI jobs and bills to Gitlab?..

**Comment by Vaserti** [comment]
Ehhh :/. it's time to use azure or gitlab

**Reply by dhvcc** [reply]
Use GitLab for enterprise, it's very good

**Comment by Viibrant** [comment]
How is this not a thing? Why isn't this high priority? 

I can't believe GitHub doesn't do this, surely this functionality is a given. Maybe I'm naive and Python is just a small toy language that is scarcely used anywhere, or maybe GitHub knows something we don't 👀

**Reply by PageotD** [reply]
..or GitLab and Azure know something GitHub doesn't...

**Reply by Viibrant** [reply]
> ..or GitLab and Azure know something GitHub doesn't...

True, GitHub small indie company

**Comment by jpablos11** [comment]
Unbelievable how this conversation and opinions have been ignored for so long, I agree Python packages is a simple and important feature.

I think I am better off looking for alternatives considering the time this thread has been active with no action, disappointed on GitHub and its "strategic direction" response, which indeed sounds to come from corporate minds out of touch with the development space.

**Reply by lukasz-mitka** [reply]
> [...] corporate minds out of touch with the development space

You've put it perfectly.

However, I'm wondering, how many paying clients have actually chosen different service because of that particular deficiency?

**Reply by psirenny** [reply]
GitLab is often chosen over GitHub for on-premise deployments because you get so many things out of the box such as Python package management. Our organization's tool of choice is GitHub, but these missing features make it a tougher sell to partners we work with that want an easy way to stand everything up.

**Reply by datalogics-kam** [reply]
We've had to resort to using CodeArtifact on AWS, which is non-ideal. Logins last 12 hours max, and are per-project, so once per work day, a developer's build will fail, and they have to remember to run the command to log into CodeArtifact.

**Reply by umm0n** [reply]
We're in the same boat @datalogics-kam. And considering how little love AWS is showing their Code* suite, I'm not thrilled about this dependency.

**Comment by scimind2460** [comment]
When is this going to be implemented? According to the TIOBE index, Python is the most popular language, with a 20.17% usage rate. I do not comprehend why GitHub does not implement a package system for Python. This is very disappointing and must be resolved. I expected that GitHub would have better integration with Python.

**Comment by jstafford5380** [comment]
I think the main issue is that GitHub has something else coming for packages to replace the current implementation.

**Reply by nearlythere** [reply]
Seems like GitHub has signalled they aren't going to develop their GitHub Packages solution - they have a partnership with JFrog https://github.blog/news-insights/product-news/github-and-jfrog/

**Reply by scimind2460** [reply]
Very interesting. I am however a bit uncomfortable with a third-party solution. I would prefer if GitHub Packages was updated. And also, could anyone please explain it to me? It's not explained very well in the article.

**Reply by scimind2460** [reply]
Also, surprisingly GitHub has not taken any action in this respect, and JFrog integration with packages still appears to not exist. Any action in the near future?

**Comment by varadinov** [comment]
I checked this few years ago and I was disappointed. Now I had the good felling that this feature will be there but now I'm even more disappointed...

**Reply by maksimu** [reply]
Same here.... :(

**Reply by dhvcc** [reply]
This is just sad

**Comment by JorenVsyntra** [comment]
Hey everyone!

**Comment by grabear** [comment]
+1

**Comment by RudyD** [comment]
LOL ... Now it's March 2025 and ChatGPT o1 Pro hallucinated and told me the feature was available. Clearly python registry support isn't here yet. In the age of AI and rocket ships that can be caught by chopsticks and self-driving cars, we still don't have a basic python registry.

Humans will colonize Mars before we get this feature. Mark my words.

Thanks GitHub, Inc.

**Reply by dhvcc** [reply]
I mean, we got gta 6 before github python package support

**Reply by faustool** [reply]
We'll get Mass Effect 4 before github python package support

**Reply by paivin-dn** [reply]
Half-Life 3 too

**Reply by scimind2460** [reply]
The Boeing 777X too

**Reply by DanielViglione** [reply]
ChatGPT o3 correctly told me it was not available.

**Comment by alacret** [comment]
This is still something that is desired, very much. We have everything but Python Packages on GitHub

**Comment by grimwm** [comment]
I'm adding my support to this feature. So far, I set up private repos for clients, but it would be nice to get it for free here.

**Reply by davidkhala** [reply]
> I'm adding my support to this feature. So far, I set up private repos for clients, but it would be nice to get it for free here.

Out of its importance, I would like to pay for it in reasonable price.

**Reply by scimind2460** [reply]
I... don't feel like paying for it is a good idea.

**Comment by MaxAnderson95** [comment]
We've had great success using [PyOCI](https://github.com/AllexVeldman/pyoci) in conjunction with GitHub Packages. It's a stateless web server that provides a PyPi compatible API, but stores and retrieves the packages as OCI artifacts in any OCI registry.

You can use the instance hosted by the creator, or run your own instance anywhere you can run a container. We run an instance on a free plan of Azure App Service.

**Comment by philomory** [comment]
I would love this as a feature, but only if they actually implement it properly; right now, the existing implementation of Maven/Gradle support is half-baked and basically useless, and GitHub has removed any improvements to it from the Road Map. If hypothetical Python support ends up getting implemented only to the same level as the Maven/Gradle support, it's probably better not to have it at all - half-implemented support just gets people to naively try using it, and then constantly fighting against all the shortcomings and having to implement a bunch of workarounds, when you'd be better off just using a different solution from the start.

In other words, if Python package support were implemented more like npm or RubyGems support, that'd be great. If it's only going to be implemented like Maven/Gradle support, then at least for me it'd be a hard pass.

**Comment by nearlythere** [comment]
Finally unsubscribing and accepting this is never happening. This was [#94](https://github.com/github/roadmap/issues/94) in their roadmap and it's "closed as not planned" --  https://github.com/github/roadmap/issues/94

> This is no longer planned due to a change in our strategic priorities and the allocation of our resources towards higher-priority initiatives.

**Comment by will-e-yams** [comment]
Snowflake built it :/
https://docs.snowflake.com/developer-guide/udf/python/udf-python-packages#artifact-repository-overview

**Comment by farridav** [comment]
Such a crucial feature, and so easy to implement, likely the product team are blissfully unaware of how many commercial decisions are made based on this missing feature, and moreover the gain in market share they would get if they implemented it.

**Comment by RempelOliveira** [comment]
Maybe they don't want this feature because they know it'll be heavily used and will require lots of dedicated resources.

This topic was opened almost 5 years ago, it's absurd.

**Comment by tbigelow-nomi** [comment]
Not having this has meant my org has moved to a competing platform instead. The number of features we get with GitHub has not kept pace with others and we're divesting in those deficient areas.

**Comment by oshriza** [comment]
2025 and python packages registry not supported yet 👎

**Comment by ken-master** [comment]
![microsoft-fuck](https://github.com/user-attachments/assets/1e5f2e25-ca9d-43b1-8db6-5bf77baa61f7)

**Comment by scimind2460** [comment]
Oddly, no official response from GitHub yet. Any way to promote this question so that it is given a higher priority?

**Comment by tonyd-stoat** [comment]
Absolute bollocks. Not implementing a feature like this for over 5 years?? really? 

I have been working with GitLab for all this time and a new project uses Enterprise GitHub (PAID FOR) and not having a feature like this.... 

Will be recommending to move the project to GitLab. Utterly disappointing.

**Reply by d-ganchar** [reply]
It’s just that five years ago there wasn’t such a crazy AI hype. Maybe Microsoft is just hoping that AI will eventually implement this feature all by itself )

**Reply by madMathematician971** [reply]
Yet here I am discovering that GitHub is owned by Microsoft. No wonder:))) I don't think it's about AI hype, the need to have this feature is really as basic as it gets for Python devs, to have a private managed registry, not all projects are open source and it just drives revenue that comes from projects that use a paid enterprise GitHub to the competition at GitLab and others. I'm surprised they implemented all other repos they currently have, but why a Python index repo that is not that hard of a system in its core is absolutely ignored is a mystery.

I'd expect Microsoft to use AI to steal people's data and sell it to advertisers rather than use it for something useful \_('<')_/

**Comment by gtskaushik** [comment]
Such a basic feature and simple to implement but no response for 5 yrs!! Huge demand means huge traffic from users once it is implemented. Probably that's the reason for not implementing it!!

**Comment by sveinse** [comment]
We are developing a lot of internal python packages which cannot be uploaded to pypi and would benefit a lot from having a org-wide python package repository. GitHub does have package registries for many other eco-systems, like npm, but not python. Alas. I'd vote +100 for getting this in GitHub. We need it.

**Comment by bartelemi** [comment]
Hey GitHub! Just a thought: maybe you could instruct Copilot Agent to add support for Python package registry to your platform 🤔 /s

**Comment by faustool** [comment]
More like in galactic star date 6202

**Comment by TugdualKerjan** [comment]
Is this a joke ?

**Reply by dhvcc** [reply]
Yeah, pretty much. But don't worry, AI will solve this somehow, surely
How? Don't ask questions 😅

**Comment by kn-sumanth** [comment]
Please!

**Comment by battlesnake** [comment]
Oh wow.  I'm sure Gitlab supported this (and a load of other registries) like 10 years ago lol

**Comment by xmaxrayx** [comment]
never saw someone recommend gh package once.

**Comment by ryewilson** [comment]
My org also needs this x  💯

**Comment by galpin** [comment]
This is really hard to fathom from a product perspective!

**Comment by samirgrl** [comment]
So much for listening to what your customers want

**Comment by aripalo** [comment]
There is a very valid use case for this: Distributing private/internal AWS CDK constructs. For AWS CDK TypeScript/JavaScript and Python are the most used languages, which means that (most) constructs need to be published both into npm- and pypi-compatible registries.

For private/internal packages, GitHub Packages is the most obvious solution. Not having Python support is really disappointing.

**Reply by MatthewCane** [reply]
I second this, we use CDK libraries for templating infrastructure patterns in Python and we have to rely on Git based package sources only which does not work with Dependabot, making maintenance much more of a chore.

**Comment by sebdanielsson** [comment]
This is long overdue @jaredpalmer 🤠

**Comment by lizhongyue** [comment]
We need to use GitHub Packages to store Python packages.What causes GitHub Packages to not support Python?

**Reply by davidkhala** [reply]
Because Microsoft has sponsored pypi, I guess

**Reply by carrete** [reply]
Microsoft owns npm.

**Comment by AliRezaei-Code** [comment]
still nothing ...

**Comment by will-e-yams** [comment]
You could try [this private-pypi package](https://pypi.org/project/private-pypi/)? 

> `private-pypi` allows you to deploy a PyPI server privately and keep your artifacts safe by leveraging the power (confidentiality, integrity and availability) of your storage backend. The backend mechanism is designed to be flexible so that the developer could support a new storage backend at a low cost.
Supported backends:
GitHub. (Yes, you can now host your Python package in GitHub by using private-pypi. )
File system.
... (Upcoming)

**Comment by MaxAnderson95** [comment]
We use pyoci and it works great. Stores the artifacts in ghcr as OCI files and provides a pypi api interface for consumption and publishing. Auth is passed thru and you can use GITHUB_TOKEN in your CI builds. You can even host your own if you want.

**Comment by iegorval** [comment]
I cannot believe that this feature is still not implemented.

**Comment by orgroman** [comment]
Even though alternatives exist such as AWS CodeArtifact JFrog Artifactory and Azure Artifacts it is still odd that Python does not have an officially supported PyPI index on GitHub unlike npm and other ecosystems...

**Comment by GrigoryEvko** [comment]
<img width="1818" height="315" alt="image" src="https://github.com/user-attachments/assets/a2b24cd5-1803-4aad-a35b-940d609b0da9" />
Priorities in 2023:
<img width="1406" height="634" alt="image" src="https://github.com/user-attachments/assets/6818ecda-942e-4245-bb53-ec36bb325614" />

**Reply by will-e-yams** [reply]
All that information above and THAT'S your post!?

<Shakes head while in hands>

My dude... 😢 I've been on this thread for a long time (definitely not as long as some). And maybe that's inconsequential to you (and certainly to GitHub PMs), but I'll tell you that you don't get action from fluffing snow off ya coat or dirt of ya shoulders.

That's, "I'm done with you" speak. 

Thumbs up some other post that sounds like your issue or make a new one, but this one you made is crushed.

I seriously believe that the only way to get this is
B/ volume: make your case know. If it's the same as another, don't clog the hopper, just upvoter.
C/ provide more _distinct_ use cases

And most importantly
A/ don't look like an internet troll. The folks that built GitHub are GenX or earlier. Take it from a gen-xer that your post is left desiring in details that matter.

0/ Gen-x or not, screenshots communicate very little. I don't think any of us are mind readers.

**Comment by NeilGraham** [comment]
Since the Github team has much more important priorities, Gitlab fully supports private PyPi packages for your organization: 
https://docs.gitlab.com/user/packages/pypi_repository/

**Comment by infakt-HNP** [comment]
Any plans to introduce Python registry?
